testtools/compat.py

```diff
@@ -26,10 +25,12 @@
 import traceback
 import unicodedata
 
-from extras import try_imports
+from extras import try_import, try_imports
 
 BytesIO = try_imports(['StringIO.StringIO', 'io.BytesIO'])
 StringIO = try_imports(['StringIO.StringIO', 'io.StringIO'])
+# To let setup.py work, make this a conditional import.
```

Comment:
Yes, unless we alter setup.py to not require importing testtools. Which I'd like to do but is a separate conceptual change. Note that it was already broken before this patch - my other changes don't introduce the damage.